lib/datadog/core/configuration/agent_settings_resolver.rb

```diff
@@ -232,7 +232,26 @@ def uds_fallback
         end
 
         def should_use_uds?
-          can_use_uds? && !mixed_http_and_uds?
+          # When we have mixed settings for http/https and uds, we print a warning
+          # and use the uds settings.
+          unless defined?(@mixed_http_and_uds)
+            @mixed_http_and_uds = (configured_hostname || configured_port) && can_use_uds?
+            if @mixed_http_and_uds
+              warn_if_configuration_mismatch(
+                [
+                  DetectedConfiguration.new(
+                    friendly_name: 'configuration for unix domain socket',
+                    value: parsed_url.to_s,
+                  ),
+                  DetectedConfiguration.new(
+                    friendly_name: 'configuration of hostname/port for http/https use',
+                    value: "hostname: '#{hostname}', port: '#{port}'",
+                  ),
+                ]
+              )
+            end
+          end
+          can_use_uds?
```

Comment:
With these changes we violate SRP. Question-mark methods doesn't suppose to set any variables, but provide only true/false checks.

Solution: Rename existing method `mixed_http_and_uds?` into `mixed_http_and_uds` and adjust `should_use_uds?` accordingly.

Reply:
The variable being set is just a cache for the value being returned or am I missing something?

Reply:
Memoizing values plays an important part when combined with the warnings: it means we don't repeat warnings just because some helper function queried the same info again.